src/core/MOM_barotropic.F90

```diff
@@ -3413,6 +3415,44 @@ subroutine set_up_BT_OBC(OBC, eta, SpV_avg, BT_OBC, BT_Domain, G, GV, US, CS, MS
   call do_group_pass(BT_OBC%pass_h, BT_Domain)
   call do_group_pass(BT_OBC%pass_cg, BT_Domain)
 
+  ! Update IareaT_OBCmask so that nothing changes outside of the OBC (problem for interior OBCs only)
+  if (BT_OBC%apply_u_OBCs) then
+    do i=isd,ied
+      if (OBC%specified_u_BCs_exist_globally .or. OBC%open_u_BCs_exist_globally) then
+        do j=jsd,jed
+          if (OBC%segnum_u(I-1,j) /= OBC_NONE) then
+            if (OBC%segment(OBC%segnum_u(I-1,j))%direction == OBC_DIRECTION_E) then
+              CS%IareaT_OBCmask(i,j) = 0.0
+            endif
+          endif
+          if (OBC%segnum_u(I,j) /= OBC_NONE) then
+            if (OBC%segment(OBC%segnum_u(I,j))%direction == OBC_DIRECTION_W) then
+              CS%IareaT_OBCmask(i,j) = 0.0
+            endif
+          endif
+        enddo
+      endif
+    enddo
+  endif
+  if (BT_OBC%apply_v_OBCs) then
+    do j=jsd,jed
+      if (OBC%specified_v_BCs_exist_globally .or. OBC%open_v_BCs_exist_globally) then
```

Comment:
The logical test on line 3439 is redundant with the test on line 3437, if you look at where `BT_OBC%apply_v_OBCs` is set up on lines 784 and 789, and hence this extra test can be removed.  The same comment applies to the logical test on line 3421.